### PR TITLE
[WIP] Feature/388: Setup a master Convict project Config 

### PR DESCRIPTION
**Pull request by ramanmangla**
This is in reference to ticket #388 .

- Replace RC with Convict
- Merge multiple configs into a master config (`src/shared/config.ts`)
- Define Convict Config definition for validation
- Auto-load config properties for dev, test and prod environments using `NODE_ENV`
- Auto-migrate DB after connection setup
- Refactor files to work with new config

### CODE REVIEW

**Comment by eoln** [comment]
@ramanmangla tests are broken

File: src/shared/config.ts [hunk 1]
```diff
@@ -20,60 +25,70 @@
...
-    COLOR?: boolean;
+  INSPECT: {
+    DEPTH: number;
+    SHOW_HIDDEN: boolean;
+    COLOR: boolean;
   };
 }
 
-const RC = parse(rc('AS', Config)) as ServiceConfig
+// const RC = parse(rc('AS', Config)) as ServiceConfig
```
Comment (eoln) [inline src/shared/config.ts:90]: drop commented code

**Review by eoln — DISMISSED** [review]
tests are failing, use Draft PR if needed

**Comment by kkzeng** [comment]
Created new branch to work on - no access to forked branch

**Pull request closed without merging**